lib/rex/proto/proxy/socks4a.rb

```diff
@@ -131,7 +131,7 @@ def is_bind?
       def resolve( hostname )
         if( not hostname.empty? )
           begin
-            return Rex::Socket.addr_itoa( Rex::Socket.gethostbyname( hostname )[3].unpack( 'N' ).first )
+            return Rex::Socket.getaddress(hostname, false)
```

Comment:
I have slightly different semantics on how this address resolves on my mac versus other machines. I opened msfconsole on this branch and within irb ran some scenarios to get different results, but interestingly on a VPN + Kali VM, they're the same results. So I assume I'm hitting different DNS resolvers between the old and new implements of rex-socket

## Hostname

```
hostname = `hostname`.chomp
Rex::Socket.addr_itoa( Rex::Socket.gethostbyname( hostname )[3].unpack( 'N' ).first )
Rex::Socket.getaddress(hostname, false)
```

My output:
```
>> hostname = `hostname`.chomp
=> "my-machine"
>> Rex::Socket.addr_itoa( Rex::Socket.gethostbyname( hostname )[3].unpack( 'N' ).first )
NOTE: Rex::Socket.gethostbyname is deprecated, use getaddress, resolve_nbo, or similar instead. It will be removed in the next Major version
=> "42.0.35.196"
>> Rex::Socket.getaddress(hostname, false)
=> "192.168.1.223"
```

## Google

```
hostname = "google.com"
Rex::Socket.addr_itoa( Rex::Socket.gethostbyname( hostname )[3].unpack( 'N' ).first )
Rex::Socket.getaddress(hostname, false)
```

My output:
```
>> hostname = "google.com"
=> "google.com"
>> Rex::Socket.addr_itoa( Rex::Socket.gethostbyname( hostname )[3].unpack( 'N' ).first )
NOTE: Rex::Socket.gethostbyname is deprecated, use getaddress, resolve_nbo, or similar instead. It will be removed in the next Major version
=> "42.0.20.80"
>> Rex::Socket.getaddress(hostname, false)
=> "216.58.205.46"
```

I'm not sure what the semantic differences are here, but I'd argue the new implementation is closer to what I'd expect.

---

Coincidentally found this blog post which seems awfully relevant:
https://blog.cyberwar.nl/2012/12/the-curious-case-of-42-0-20-80/

Specifically:
>```
> $ host -t aaaa www.google.com
> www.google.com has IPv6 address 2a00:1450:4013:c00::63
> ```
> ….but I only now noticed that the first four bytes of that address, 2a00:1450, hexadecimally represent 42.0.20.80!

Which aligns with the above behavior I was seeing